packages/p2p/src/components/modal-manager/modals/quick-add-modal/quick-add-modal.tsx

```diff
@@ -0,0 +1,174 @@
+import React from 'react';
+import classNames from 'classnames';
+import { DesktopWrapper, Icon, MobileFullPageModal, MobileWrapper, Modal } from '@deriv/components';
+import { observer } from '@deriv/stores';
+import { useStores } from 'Stores';
+import { localize } from 'Components/i18next';
+import { useModalManagerContext } from 'Components/modal-manager/modal-manager-context';
+import { buy_sell } from 'Constants/buy-sell';
+import { TAdvertProps } from 'Types/adverts.types';
+import QuickAddModalContent from './quick-add-modal-content';
+import QuickAddModalButtons from './quick-add-modal-buttons';
+
+type TQuickAddModalProps = {
+    advert: Partial<TAdvertProps>;
+};
+
+const QuickAddModal = ({ advert }: TQuickAddModalProps) => {
+    const { my_ads_store, my_profile_store } = useStores();
+    const {
+        should_show_add_payment_method,
+        show_filter_payment_methods,
+        hideQuickAddModal,
+        onClickUpdatePaymentMethods,
+        setSearchTerm,
+        setSearchedResults,
+        setShouldShowAddPaymentMethod,
+        setShowFilterPaymentMethods,
+    } = my_ads_store;
+    const { selected_payment_method, getPaymentMethodsList } = my_profile_store;
+
+    const { is_modal_open, showModal, useSavedState } = useModalManagerContext();
+
+    const type = advert ? advert.type : null;
```

Comment:
```suggestion
```

Reply:
We only check advert.type once so i just think we dont need to put it in a variable